File: elovate_backend/src/main/java/com/tuwien/elovate/dtos/gameapplication/GameApplicationResponseDto.java
package com.tuwien.elovate.dtos.gameapplication;

/*-
 * #%L
 * ELOvate
 * %%
 * Copyright (C) 2024 ELOvate GmbH.
 * %%
 * Copyright (C) 2024 ELOvate GmbH. - All Rights Reserved
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * #L%
 */

import com.tuwien.elovate.enums.GameApplicationStatus;
import com.tuwien.elovate.enums.Genre;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@Data
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
public class GameApplicationResponseDto {

    private Long id;

    private String name;

    private Genre genre;

    private boolean drawPossible;

    private Integer playersPerTeam;

    private GameApplicationStatus status;

    private Long developerId;

    private String image;

    private Long gameId;

}

File: elovate_frontend/src/dtos/GameApplication/gameApplicationResponseDto.ts
/*-
 * #%L
 * ELOvate
 * %%
 * Copyright (C) 2024 ELOvate GmbH.
 * %%
 * Copyright (C) 2024 ELOvate GmbH. - All Rights Reserved
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * #L%
 */
import {Genre} from "../enums/genre";
import {GameApplicationStatus} from "../enums/gameApplicationStatus";

export interface GameApplicationResponseDto {
  id: number,
  name: string,
  genre: Genre,
  drawPossible: boolean,
  playersPerTeam: number,
  image: string | null,
  status: GameApplicationStatus,
  developerId: number;
  gameId: number;
}
